cvmfs/statistics_database.h

```diff
@@ -0,0 +1,110 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_STATISTICS_DATABASE_H_
+#define CVMFS_STATISTICS_DATABASE_H_
+
+#include <ctime>
+#include <string>
+
+#include "options.h"
+#include "sql.h"
+#include "swissknife.h"
+#include "util/posix.h"
+
+typedef struct {
+  std::string files_added;
+  std::string files_removed;
+  std::string files_changed;
+  std::string dir_added;
+  std::string dir_removed;
+  std::string dir_changed;
+  std::string bytes_added;
+  std::string bytes_removed;
+} Stats;
+
+struct RevisionFlags {
+  enum T {
+    kInitialRevision   = 1,
+    kUpdatableRevision = 2,
+    kUpdatedRevision   = 3,
+    kFailingRevision   = 4,
+  };
+};
+
+class StatisticsDatabase : public sqlite::Database<StatisticsDatabase> {
+ public:
+  // not const - needs to be adaptable!
+  static float        kLatestSchema;
+  // not const - needs to be adaptable!
+  static unsigned     kLatestSchemaRevision;
+  static const float  kLatestCompatibleSchema;
+  static bool         compacting_fails;
+  static unsigned int  instances;
+  unsigned int         create_empty_db_calls;
+  unsigned int         check_compatibility_calls;
+  unsigned int         live_upgrade_calls;
+  mutable unsigned int compact_calls;
+
+  bool CreateEmptyDatabase();
+  bool CheckSchemaCompatibility();
+  bool LiveSchemaUpgradeIfNecessary();
+  bool CompactDatabase() const;
+  ~StatisticsDatabase();
+
+/**
+  * Get command statistics values and convert them into strings.
+  *
+  * @param command to access counter statistics
+  * @return a Stats struct with all statistics values stored in strings
+  */
+  Stats GetStats(swissknife::Command *command);
```

Comment:
Here and in `StoreStatistics` at line 84, is it possible to to take the parameter as a `const swissknife::Command*` or `const swissknife::Command&`? 

It may require adding `const` to the Lookup method of the `Statistics` class.